Integrations/integration-VirusTotal-Private_API.yml

```diff
@@ -551,98 +553,120 @@ script:
         """
 
         args = demisto.args()
-        url = args.get('resource')
+        urls = argToList(args.get('resource'))
         all_info = 1 if args.get('allInfo', None) == 'true' else 0
         threshold = int(args.get('threshold', None) or demisto.params().get('urlThreshold', None) or 10);
         short_format = args.get('shortFormat', None) == 'true'
         full_response = args.get('fullResponse', None) == 'true'
+        retries_left = int(args.get('retries', 2))
         if (full_response):
             max_len = 1000
         else:
             max_len = 50
 
-        response = get_url_report(url, all_info)
-
-        if (response.get('response_code', None) == -1):
-            return "Invalid url"
+        completed_responses = []
+        completed_responses_dict = []
+        non_completed_urls = []
+        for url in urls:
+            response = get_url_report(url, all_info)
 
-        positives = demisto.get(response, 'positives')
+            if (response.get('response_code', None) == -1):
+                return "Invalid url"
 
-        md = '## VirusTotal URL report for: ' + url + '\n'
-        md += 'Scan ID: **' + str(response.get('scan_id', '')) + '**\n'
-        md += 'Scan date: **' + str(response.get('scan_date', '')) + '**\n'
-        md += 'Detections / Total: **' + str(positives) + '/' + str(response.get('total', '')) + '**\n'
-        md += 'VT Link: [' + str(response.get('resource', '')) + '](' + str(response.get('permalink', '')) + ')\n'
-
-        ec = {}
-        ec['DBotScore'] = []
-        dbotScore = 0
+            if (not response.get('total')):
+                non_completed_urls.append(url)
+            else:
+                completed_responses_dict.append({
+                    'response': response,
+                    'url': url
+                })
+                completed_responses.append(response)
+
+        while non_completed_urls and retries_left:
+            retries_left -= 1
+            # In case there were url scans that have not finished: try again after giving them enough time to finish
+            time.sleep(SCAN_FINISH_TIME_IN_SECONDS)
+            for url in non_completed_urls:
+                response = get_url_report(url, all_info)
+                completed_responses_dict.append({
+                    'response': response,
+                    'url': url
+                })
+                completed_responses.append(response)
+        md = ''
+        ec = {
+            'DBotScore': [],
+            outputPaths['url']: [],
 
-        if (positives >= threshold):
-            ec.update({
-                outputPaths['url']: {
+        }
+        for response_dict in completed_responses_dict:
```

Comment:
maybe it is better to create a function that handle a single url and creates the context output and part of the human readable and call it in a loop. rather than have a loop a lot of code in it.

Reply:
It's possible, but that would require a single function to handle these flows:
1. Get scan results from endpoint.
2. Support the retry mechanism.
3. Get MD + EC results.

It's possible, but it's a bit different than the flow of other functions.